Kernel/Arch/i386/CPU.cpp

```diff
@@ -133,7 +137,7 @@ static void dump(const DumpType& regs)
         esp = regs.esp_if_crossRing;
     }
 
-    if constexpr (IsSame<DumpType, RegisterDumpWithExceptionCode>::value) {
+    if constexpr (IsSame<DumpType, RegisterDump>::value) {
```

Comment:
Since this patch gets rid of `RegisterDumpWithExceptionCode`, there's no longer any need for the templatized `dump()` function here.
It can just be `dump(const RegisterDump&)` now :)